File: problem/1/python/solve_1.py
#!/usr/bin/env python
# http://projecteuler.net/problem=1

import sys

upper_limit = int(sys.argv[1])
multipliers = [3, 5]
results = []

for x in multipliers:
        for y in range(1, upper_limit + 1):
                res = y * x
                if res < upper_limit:
                        results.append(res)

print sum(list(set(results)))

File: problem/1/perl/solve_1.pl
#!/usr/bin/env perl6
# http://projecteuler.net/problem=1

use v6;

die "Exactly one argument needed (natural number)." unless @*ARGS;

my $upper_limit = @*ARGS[0];
my @multipliers = (3, 5);
my @results;

for @multipliers -> $m {
        loop (my $i = 1; $i * $m < $upper_limit; $i++) {
                @results.push($i * $m);
        }
}

say [+] @results.uniq;
